aws-logging-cloudwatch/src/main/java/com/amplifyframework/logging/cloudwatch/AWSCloudWatchLoggingPlugin.kt

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amplifyframework.logging.cloudwatch
+
+import android.content.Context
+import aws.sdk.kotlin.services.cloudwatchlogs.CloudWatchLogsClient
+import com.amplifyframework.AmplifyException
+import com.amplifyframework.auth.CognitoCredentialsProvider
+import com.amplifyframework.core.Action
+import com.amplifyframework.core.Amplify
+import com.amplifyframework.core.Consumer
+import com.amplifyframework.core.category.CategoryType
+import com.amplifyframework.logging.Logger
+import com.amplifyframework.logging.LoggingPlugin
+import com.amplifyframework.logging.cloudwatch.models.AWSCloudWatchLoggingPluginConfiguration
+import java.net.URL
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import org.json.JSONObject
+
+/**
+ * Logging plugin to send logs to AWS Cloudwatch
+*/
+class AWSCloudWatchLoggingPlugin @JvmOverloads constructor(
+    private val awsCloudWatchLoggingPluginConfig: AWSCloudWatchLoggingPluginConfiguration? = null,
+    private val awsRemoteLoggingConstraintProvider: RemoteLoggingConstraintProvider? = null,
+) : LoggingPlugin<CloudWatchLogsClient>() {
+
+    private val loggingConstraintsResolver =
+        LoggingConstraintsResolver(
+            localLoggingConstraint = awsCloudWatchLoggingPluginConfig?.loggingConstraints,
+            remoteLoggingConstraintProvider = awsRemoteLoggingConstraintProvider,
+        )
+    private val awsCloudWatchLoggingPluginImplementation = AWSCloudWatchLoggingPluginImplementation(
+        loggingConstraintsResolver,
+        awsCloudWatchLoggingPluginConfig,
+    )
+    private lateinit var cloudWatchLogsClient: CloudWatchLogsClient
+    private val logger = Amplify.Logging.logger(CategoryType.LOGGING, this::class.java.simpleName)
+
+    companion object {
+        private const val CONFIG_FILENAME = "amplifyconfiguration_logging"
+        internal const val SHARED_PREFERENCE_FILENAME = "com.amplify.logging.a3fa4188-0ac5-11ee-be56-0242ac120002"
+    }
+
+    @Deprecated("Deprecated in Java")
+    override fun forNamespace(namespace: String?): Logger {
+        return awsCloudWatchLoggingPluginImplementation.forNamespace(namespace)
```

Comment:
This should directly call `logger(namespace)` and we can remove `forNamespace` from AWSCloudWatchLoggingPluginImplementation.